software/debugging/index.rst

```diff
@@ -0,0 +1,456 @@
+
+.. tabbed:: Frontier
+
+    GDB is availableon Summit under all compiler families:
+
+    .. code::
+
+        module load gdb
+
+    To use GDB to debug your application run:
+
+    .. code::
+
+        gdb ./path_to_executable
+
+    dditional information about GDB usage can befound on the `GDB Documentation Page <https://www.sourceware.org/gdb/documentation/>`__.
+
+
+********
+Valgrind
+********
+
+.. tabbed:: Summit
+
+    `Valgrind <http://valgrind.org>`__ is an instrumentation framework for
+    building dynamic analysis tools. There are Valgrind tools that can
+    automatically detect many memory management and threading bugs, and
+    profile your programs in detail. You can also use Valgrind to build new
+    tools.
+
+    The Valgrind distribution currently includes five production-quality
+    tools: a memory error detector, a thread error detector, a cache and
+    branch-prediction profiler, a call-graph generating cache profiler,
+    and a heap profiler. It also includes two experimental tools: a data
+    race detector, and an instant memory leak detector.
+
+    The Valgrind tool suite provides a number of debugging and
+    profiling tools. The most popular is Memcheck, a memory checking tool
+    which can detect many common memory errors such as:
+
+        - Touching memory you shouldn’t (eg. overrunning heap block boundaries, or reading/writing freed memory).
+        - Using values before they have been initialized.
+        - Incorrect freeing of memory, such as double-freeing heap blocks.
+        - Memory leaks.
+
+    Valgrind is available on Summit under all compiler families:
+
+    .. code::
+
+        module load valgrind
+
+    Additional information about Valgrind usage and OLCF-provided builds can
+    be found on the `Valgrind Software Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
+
+.. tabbed:: Andes
+
+    `Valgrind <http://valgrind.org>`__ is an instrumentation framework for
+    building dynamic analysis tools. There are Valgrind tools that can
+    automatically detect many memory management and threading bugs, and
+    profile your programs in detail. You can also use Valgrind to build new
+    tools.
+
+    The Valgrind distribution currently includes five production-quality
+    tools: a memory error detector, a thread error detector, a cache and
+    branch-prediction profiler, a call-graph generating cache profiler,
+    and a heap profiler. It also includes two experimental tools: a data
+    race detector, and an instant memory leak detector.
+
+    The Valgrind tool suite provides a number of debugging and
+    profiling tools. The most popular is Memcheck, a memory checking tool
+    which can detect many common memory errors such as:
+
+        - Touching memory you shouldn’t (eg. overrunning heap block boundaries,
+        or reading/writing freed memory).
```

Comment:
Line 427 needs to be indented 2 more spaces (to make it flush with what is above) so that the list prints out correctly.

Reply:
fixed in 15e4ea35ac8e6c884ee1e03be22a3c76c1d214d3